appinventor/appengine/src/com/google/appinventor/client/DesignToolbar.java

```diff
@@ -386,7 +387,12 @@ public void execute() {
       }
       // Only do something if we aren't already doing it!
       if (!lockPublishButton) {
-        lockPublishButton = true;
+        lockPublishButton = false;
+        if (Ode.getInstance().getCurrentYoungAndroidProjectRootNode().hasExtensions()){
+          Window.alert("cant publish project with extensions");
```

Comment:
Move the if statement before `lockPublishButton = true;` and then return after the alert. Otherwise, this will fall through and still try to send to the gallery.